Library/Homebrew/utils/curl.rb

```diff
@@ -461,7 +461,10 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default], r
         url.start_with?("http://") &&
         secure_details[:final_url].present? && secure_details[:final_url].start_with?("https://")
 
-      if (etag_match || content_length_match || file_match) && http_with_https_available
+      # curl dependencies are required to use http mirrors, so we should skip this check
+      curl_dep = self.class.curl_deps.include?(owner.name)
```

Comment:
Instead the audit itself should be skipped rather than adding conditional logic there.